linkerd/app/src/lib.rs

```diff
@@ -134,7 +134,7 @@ impl Config {
         let (metrics, report) = Metrics::new(admin.metrics_retain_idle);
 
         debug!("Building DNS client");
-        let dns = dns.build();
+        let dns = dns.build(&mut registry);
```

Comment:
and then we can use:

`
```suggestion
        let dns = dns.build(registry.sub_registry_with_prefix("control_dns"));
```

This helps to scope the dns client like the other controller related clients.

Reply:
see [above](https://github.com/linkerd/linkerd2-proxy/pull/3822#discussion_r2023111280), this is addressed in 1437755.